internal/core/src/common/Utils.h

```diff
@@ -205,4 +205,21 @@ Join(const std::vector<T>& items, const std::string& delimiter) {
     return ss.str();
 }
 
+inline std::string
+GetCommonPrefix(const std::string& str1, const std::string& str2) {
+    std::string result;
```

Comment:
how about  reserve( len )  ?

Reply:
not necessary, it is fast enough

Reply:
or you can first calcuate the len and return by str1.sbustr(0, len)

Reply:
size_t len = std::min(str1.length(), str2.length());
size_t i = 0;
while(i<len && str1[i] == str2[i]) ++i;
return str1.substr(0, i);

Reply:
your implement is same if considering performance and resource use. it is  just more concise.